src/Dialect/Mlir/DialectBuilder.cpp

```diff
@@ -963,6 +976,59 @@ memref::DeallocOp MemRefBuilder::dealloc(Value val) const {
   return b().create<memref::DeallocOp>(loc(), val);
 }
 
+//===----------------------------------------------------------------------===//
+// Reshape.
+
+memref::ReshapeOp MemRefBuilder::reshape(
+    MemRefType destType, Value valToReshape, Value destShapeStoredInMem) const {
+  return b().create<memref::ReshapeOp>(
+      loc(), destType, valToReshape, destShapeStoredInMem);
+}
+
+memref::ReshapeOp MemRefBuilder::reshapeToFlat(Value valToReshape,
+    llvm::SmallVectorImpl<IndexExpr> &dims, Value &size1D) const {
```

Comment:
It looks to me `dims` is redundant. Are they the dims of `valToReshape`?

Reply:
yes, but since they might be already computed, I did not want to insert code to recompute them inside the call. I can make a second call without it if you prefer.

Reply:
I see, that's fine as it is. Thanks!